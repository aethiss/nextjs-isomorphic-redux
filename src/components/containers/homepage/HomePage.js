import React, { Component } from 'react'
import PropTypes from 'prop-types'

// Next
import Link from 'next/link'

// Styles
import styles from '../../../../styles/Home.module.css'

class HomePage extends Component {
  render() {
    return (
      <main className={styles.main}>
        <h3 className={styles.title}>
          Aethiss Isomorphic App with <a href='https://nextjs.org'>Next.js!</a>
        </h3>

        <div className={styles.grid}>
          <Link href='/SSR'>
            <a className={styles.card}>
              <h3>SSR Page &rarr;</h3>
              <p>Example of Server Side Render Page</p>
            </a>
          </Link>

          <Link href='/CSR'>
            <a className={styles.card}>
              <h3>CSR Page &rarr;</h3>
              <p>Example of Client Side Render Page (No SEO)</p>
            </a>
          </Link>

          <a href='https://nextjs.org/docs' className={styles.card}>
            <h3>Documentation &rarr;</h3>
            <p>Find in-depth information about Next.js features and API.</p>
          </a>

          <a
            href='https://github.com/vercel/next.js/tree/master/examples'
            className={styles.card}
          >
            <h3>Examples &rarr;</h3>
            <p>Discover and deploy boilerplate example Next.js projects.</p>
          </a>
        </div>
      </main>
    )
  }
}

export default HomePage
