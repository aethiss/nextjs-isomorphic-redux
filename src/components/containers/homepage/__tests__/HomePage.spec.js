import React from 'react'
import HomePage from '../HomePage'
import TestRenderer from 'react-test-renderer'

describe('HomePage Component', () => {
  describe('Render', () => {
    it('[HomePage] Default render desktop', () => {
      const testRenderer = TestRenderer.create(<HomePage />)
      expect(testRenderer).toMatchSnapshot()
    })
  })
})
