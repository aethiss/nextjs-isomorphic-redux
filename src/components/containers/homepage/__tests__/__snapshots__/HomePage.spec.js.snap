// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Component Render [HomePage] Default render desktop 1`] = `
<main
  className="main"
>
  <h3
    className="title"
  >
    Aethiss Isomorphic App with 
    <a
      href="https://nextjs.org"
    >
      Next.js!
    </a>
  </h3>
  <div
    className="grid"
  >
    <a
      className="card"
      href="/SSR"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <h3>
        SSR Page →
      </h3>
      <p>
        Example of Server Side Render Page
      </p>
    </a>
    <a
      className="card"
      href="/CSR"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <h3>
        CSR Page →
      </h3>
      <p>
        Example of Client Side Render Page (No SEO)
      </p>
    </a>
    <a
      className="card"
      href="https://nextjs.org/docs"
    >
      <h3>
        Documentation →
      </h3>
      <p>
        Find in-depth information about Next.js features and API.
      </p>
    </a>
    <a
      className="card"
      href="https://github.com/vercel/next.js/tree/master/examples"
    >
      <h3>
        Examples →
      </h3>
      <p>
        Discover and deploy boilerplate example Next.js projects.
      </p>
    </a>
  </div>
</main>
`;
